carma-messenger-core/cpp_message/src/cpp_message.cpp

```diff
@@ -85,6 +88,14 @@ namespace cpp_message
             Mobility_Response decode;
             auto output=decode.decode_mobility_response_message(array);
             mobility_response_message_pub_.publish(output);
+        }
+                else if(msg->messageType=="MobilityPath")   
+        {
+            std::vector<uint8_t> array=msg->content;
+            Mobility_Path decode;
+            auto output=decode.decode_mobility_path_message(array);
+            mobility_path_message_pub_.publish(output.get());
```

Comment:
You need to check if the decoding was successful. Otherwise you'll get a segfault on the publication.

Reply:
Updated